File: Examples/VisualBasic/GroupDocs.Conversion.Examples.VisualBasic/ConversionManager.vb
'ExStart:ConversionManagerWithInterfaceClass
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.IO
Imports GroupDocs.Conversion.Config
Imports GroupDocs.Conversion.Converter.Option
Imports GroupDocs.Conversion.Handler

Namespace GroupDocsConversionExamples.VisualBasic
    Public NotInheritable Class ConversionManager
        Implements IConversionProgressListener
        Implements IConversionStatusListener
        Private ReadOnly _conversionHandler As ConversionHandler
        Public Sub New(path As String)
            _conversionHandler = Common.getConversionHandler()
            _conversionHandler.SetConversionProgressListener(Me)
            _conversionHandler.SetConversionStatusListener(Me)
        End Sub
        Public Sub ConversionProgressChanged(args As ConversionProgressEventArgs) Implements IConversionProgressListener.ConversionProgressChanged
            Console.WriteLine("Conversion progress: {0} %", args.Progress)
        End Sub
        Public Sub ConversionStatusChanged(args As ConversionEventArgs) Implements IConversionStatusListener.ConversionStatusChanged
            Console.WriteLine("Conversion status changed to: {0}", args.Status)
        End Sub
        Public Function Convert(file As String) As String
            Dim options As New PdfSaveOptions()
            options.OutputType = OutputType.String
            Return _conversionHandler.Convert(Of String)(file, options)
        End Function
    End Class
End Namespace
'ExEnd:ConversionManagerWithInterfaceClass

File: Examples/CSharp/GroupDocs.Conversion.Examples.CSharp/ConversionManager.cs
//ExStart:ConversionManagerWithInterfaceClass
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.IO;
using GroupDocs.Conversion.Config;
using GroupDocs.Conversion.Handler;
using GroupDocs.Conversion.Options.Save;

namespace GroupDocs.Conversion.Examples.CSharp
{
    public class ConversionManager : IConversionProgressListener, IConversionStatusListener
    {
        private readonly ConversionHandler _conversionHandler;
        public ConversionManager(string path)
        {
            _conversionHandler = Common.getConversionHandler();
            _conversionHandler.SetConversionProgressListener(this);
            _conversionHandler.SetConversionStatusListener(this);
        }
        public void ConversionProgressChanged(ConversionProgressEventArgs args)
        {
            Console.WriteLine("Conversion progress: {0} %", args.Progress);
        }
        public void ConversionStatusChanged(ConversionEventArgs args)
        {
            Console.WriteLine("Conversion status changed to: {0}", args.Status);
        }
        public string Convert(string file)
        {
            
            var convertedDocumentPath =_conversionHandler.Convert(file, new PdfSaveOptions());
            convertedDocumentPath.Save("result-" + Path.GetFileNameWithoutExtension(Common.inputGUIDFile) + ".pdf");
            return "result-" + Common.inputGUIDFile + ".pdf";
        }
    }
}
//ExEnd:ConversionManagerWithInterfaceClass
